src/test/ensemble_flow_custom_node_tests.cpp

```diff
@@ -1981,6 +1981,39 @@ TEST_F(EnsembleConfigurationValidationWithCustomNode, SuccessfulConfiguration) {
     ASSERT_EQ(pipelineDefinition->validate(manager), StatusCode::OK);
 }
 
+TEST_F(EnsembleConfigurationValidationWithCustomNode, SuccessfulConfigurationWithDynamicShape) {
```

Comment:
Can we have a simpler test that would test new method - isConnectionShapeCorrect directly?
I assume it might be used elsewhere so it would be nice to have test just for this method and not rely only on `pipelineDefinition->validate()` call.